sponsorship_compassion/model/contracts.py

```diff
@@ -48,11 +48,23 @@ def _get_contract_from_invoice(invoice_obj, cr, uid, invoice_ids,
         for invoice in invoice_obj.browse(cr, uid, invoice_ids, ctx):
             if invoice.state == 'paid':
                 self._invoice_paid(cr, uid, invoice, ctx)
-                last_pay_date = max([move_line.date
-                                     for move_line in invoice.payment_ids
-                                     if move_line.credit > 0] or [0])
+
+                pay_dates = [move_line.date
+                             for move_line in invoice.payment_ids
+                             if move_line.credit > 0] or [0]
+
+                last_pay_date = max(pay_dates)
+                first_pay_date = min(pay_dates)
+
+                self._cancel_invoices(

```

Comment:
If you call cancel_invoices from here, you will call it each time an invoice is paid, which is not correct. We want to call it only if a contract is activated.